src/colmap/estimators/alignment.cc

```diff
@@ -441,8 +439,7 @@ bool MergeReconstructions(const double max_reproj_error,
     src_image.CamFromWorld() =
         TransformCameraWorld(tgt_from_src, src_image.CamFromWorld());
     if (!tgt_reconstruction.ExistsCamera(src_image.CameraId())) {
-      tgt_reconstruction.AddCamera(
-          src_reconstruction.Camera(src_image.CameraId()));
+      tgt_reconstruction.AddCamera(*src_image.CameraPtr());
     }
     tgt_reconstruction.AddImage(src_image);
```

Comment:
Is this correct? The camera was copied in AddCamera so the address has changed?

Reply:
Actually thinking of this, the ``AddImage`` method from https://github.com/colmap/colmap/pull/2762 is potentially dangerous. It should reset the camera pointer to the one for the camera in the reconstruction. The current check does not handle the life of the camera in a safe way.

Reply:
Ah OK I missed the update of the implementation. My design before was buggy but the new is correct. Never mind : )

Reply:
Thanks for catching this bug. Also, the function should have a unit test, though it's beyond the scope of this PR now. I added a TODO to keep track of it.

Reply:
Added here: https://github.com/colmap/colmap/pull/2766.